src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -50,12 +50,21 @@ public LargeArrayBuilder(bool initialize)
         /// Use <see cref="Add"/> if adding to the builder is a bottleneck for your use case.
         /// Otherwise, use <see cref="SlowAdd"/>.
         /// </remarks>
+        public void Add(T item) => Add(item, limit: int.MaxValue);
+
+        /// <summary>
+        /// Adds an item to this builder with the specified limit.
+        /// </summary>
+        /// <param name="item">The item to add.</param>
+        /// <param name="limit">The maximum capacity to allocate.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void Add(T item)
+        public void Add(T item, int limit)
```

Comment:
> I have been thinking of using it in Reverse, for example

Where would it need to be stored in a field?  If you mean using it instead of using Buffer and storing the array, you're likely going to run into other complications.

> Future-proofing

While it's good to think about future usage, from my perspective this change makes existing code harder to understand, unnecessarily.  We should only do so if/when there's known good reason to do so.